proto/metalstack/api/v2/switch.proto

```diff
@@ -0,0 +1,198 @@
+syntax = "proto3";
+
+package metalstack.api.v2;
+
+import "buf/validate/validate.proto";
+
+// Switch represents a network switch
+message Switch {
+  // ID of the switch
+  string id = 1 [(buf.validate.field).string = {
+    hostname: true
+    min_len: 2
+    max_len: 128
+  }];
+  // Description of the switch
+  string description = 2 [(buf.validate.field).string = {
+    min_len: 2
+    max_len: 128
+  }];
+  // Rack ID if the switch resides in a rack
+  optional string rack_id = 3 [(buf.validate.field).string = {
+    min_len: 2
+    max_len: 128
+  }];
+  // Partition ID of the partition the switch belongs to
+  string partition_id = 4 [(buf.validate.field).string = {
+    min_len: 2
+    max_len: 128
+  }];
+  // Type signifies what role a switch has, e.g. whether it is a leaf switch, or a spine, etc.
+  SwitchType type = 5 [(buf.validate.field).enum.defined_only = true];
+  // Replace mode is used to mark a switch ready for replacement
+  SwitchReplaceMode replace_mode = 6 [(buf.validate.field).enum.defined_only = true];
+  // Management IP is the switch's IP for management access
+  string management_ip = 7 [(buf.validate.field).string.ip = true];
+  // Management user is the user name to use for management access
+  string management_user = 8 [(buf.validate.field).string = {
+    min_len: 2
+    max_len: 128
+  }];
+  // Console command is the command for accessing the switch's console
+  string console_command = 9 [(buf.validate.field).string = {
+    min_len: 2
+    max_len: 128
+  }];
+  // Switch NICs are the front panel ports of the switch
+  repeated SwitchNic switch_nics = 10;
+}
+
+// SwitchOS holds information about the NOS and versions running on the switch
+message SwitchOS {
+  // Switch OS vendor identifies what NOS distribution is running on the switch, e.g. SONiC
+  SwitchOSVendor vendor = 1 [(buf.validate.field).enum.defined_only = true];
+  // Version specifies what NOS version is currently installed on the switch
+  string version = 2 [(buf.validate.field).string = {
+    min_len: 2
+    max_len: 128
+  }];
+  // Metal core version is the currently running version of the metal-core
+  string metal_core_version = 3 [(buf.validate.field).string = {
+    min_len: 2
+    max_len: 128
+  }];
+}
+
+// SwitchNic represents a front panel port and its configuration
+message SwitchNic {
+  // Name of the switch port
+  string name = 1 [(buf.validate.field).string = {
+    min_len: 2
+    max_len: 128
+  }];
+  // Identifier of the port
+  string identifier = 2 [(buf.validate.field).string = {
+    min_len: 2
+    max_len: 128
+  }];
+  // MAC address of the port
+  string mac_address = 3 [(validate.rules).string.pattern = "^([0-9A-Fa-f]{2}[:]){5}([0-9A-Fa-f]{2})$"];
```

Comment:
Please add a test for this validation in go/test/validation/switch_test.go

Reply:
The extension isn't found anyway. Using regular string length validation.